contributors/design-proposals/autoscaling/hpa-step-scaling.md

```diff
@@ -0,0 +1,143 @@
+against the upper and lower bounds defined by the step adjustments to determine which step adjustment 
+to perform.
+
+A concrete example for step scaling policy is:
+
+* Lower bound: NULL, Upper bound: 20, Adjustment: -2; 
+* Lower bound: 20, Upper bound: 40, Adjustment: -1;
+* Lower bound: 60, Upper bound: 80, Adjustment: +1;
+* Lower bound: 80, Upper bound: NULL, Adjustment: +2;
+
+With this policy, you are maintaining the Utilization in [40, 60]:
+
+* if utilization is lower than 40% but higher than 20%, decrease the replica by 1
+* If utilization is lower than 20%, decrease the replica by 2
+* if utilization is higher than 60% but lower than 80%, increase the replica by 1
+* if utilization is higher than 80%, increase the replica by 2
+
+There are a few rules for the step adjustments for your policy:
+
+* The ranges of your step adjustments can't overlap or have a gap.
+* Only one step adjustment can have a null lower bound (negative infinity). If one step adjustment has 
+    a negative lower bound, then there must be a step adjustment with a null lower bound.
+* Only one step adjustment can have a null upper bound (positive infinity). If one step adjustment has 
+    a positive upper bound, then there must be a step adjustment with a null upper bound.
+* The upper and lower bound can't be null in the same step adjustment.
+* If the metric value is above the breach threshold, the lower bound is inclusive and the upper bound is exclusive. If the metric value is below the breach threshold, the lower bound is exclusive and the upper bound is inclusive.
+
+## Scenarios where step scaling policy performs better
+
+### Scenrios:
+
+##### Pod capacity
+
+Assume we have an application, one instance of the applicaiton  can serve 10 request per second with 100% CPU
+utilization. 
+
+##### Request pattern
+At the very beginning, clients send 5 requests per second, then suddently begin to send 80 requests per second
+
+##### HPA parameters
+*  min pod num: 1
+*  max pod num: 20
+
+##### Target
+We want to achieve high resource utilization as well as scale fast.
+
+### Target Tracking Scaling with TargetCPUUtilization = 80%
+
+scaling policy: because we want to have high resources utilization, so we set target cpu utilization as 80%
+
+scaling steps:
+
+* round 1: 5 requests per seconds, CurrentReplicas = 1
+* roubd 2: 80 requests per seconds, CurrentReplicas = 1, DesiredReplicas = ceil(100/80) = 2
+* round 3: 80 requests per seconds, CurrentReplicas = 2, DesiredReplicas = ceil(200/80) = 3
+* round 4: 80 requests per seconds, CurrentReplicas = 3, DesiredReplicas = ceil(300/80) = 4
+* round 5: 80 requests per seconds, CurrentReplicas = 4, DesiredReplicas = ceil(400/80) = 5
+* round 6: 80 requests per seconds, CurrentReplicas = 5, DesiredReplicas = ceil(500/80) = 7
+* round 7: 80 requests per seconds, CurrentReplicas = 7, DesiredReplicas = ceil(700/80) = 9
+* round 8: 80 requests per seconds, CurrentReplicas = 9, DesiredReplicas = ceil(800/80) = 10
```

Comment:
this section does not actually describe the current behavior of the HPA at all (see my above comment)